packages/popover/src/index.vue

```diff
@@ -45,6 +45,10 @@ export default defineComponent({
       type: Boolean,
       default: true,
     },
+    tabindex: {
+      type: Number,
+      default: 0,
+    },
```

Comment:
I think we do not default need this, default should be `undefined` since adding this attribute to a DOM element will make the element `keyboard navigable`

Reply:
大佬你的意思是不给默认值了是吧
```js
props: {
  // 其他省略
  tabindex: Number
}

```
那官网文档需要修改吗？

![image](https://user-images.githubusercontent.com/20411966/111319285-c997f300-86a0-11eb-89c9-cc5e9fc972ce.png)

Reply:
```js
props: {
  tabindex: { 
    type: Number,
    default: undefined,
  }
}
```
Change the documentation from 0 to `-`.
文档改成 `-` 这样吧，记得每个语言都要改一下

Reply:
好的，大佬。(明天我改下，家里电脑代码拉不下来)


我记得vue2组件的props选项中
```js
props:{
  tabindex: Number
}
```
用户使用组件时，如果不传tabindex属性, 这样组件实例的tabindex应该强等于undefined，难道vue3中变了？一定要这么写了？
```js
props: {
  tabindex: { 
    type: Number,
    default: undefined,
  }
}
```
明天我追下vue3解析props的相关源码。

感谢review代码

Reply:
大佬，我追了下vue3的部分代码

```js
写法一:
props:{
  tabindex: Number
}

写法二:
props: {
  tabindex: { 
    type: Number,
    default: undefined,
  }
}
```

这2者,从表现结果上看没啥区别，但内部还是有细微的区别

表现结果没啥区别的原因:

ElPopper组件创建实例后，访问tabindex属性，如果用户不传这个属性，就是undefined。
当tabindex属性传给trigger组件时,最终安装元素时会 hostPatchProp -> patchAttr(该函数有个判断 当值为null或者undeifned时会移除该属性，否则添加该属性),所以上面2种写法都可以。

区别(就是上面2种写法为啥 当用户不传tabindex属性时，ElPopper组件实例的tabindex的值都为undefined):

写法一:
ElPopper组件实例 instance.props 没有 tabindex属性 
当在ElPopper组件的setup函数中访问this.tabindex,这里的this是指创建组件实例对象中的ctx对象,ctx被代理了,访问tabindex属性，最终会访问如下
(instance.props || {}).tabindex === undefined

写法二:
ElPopper组件实例 instance.props 有 tabindex属性 
当在ElPopper组件的setup函数中访问this.tabindex,最终会被访问如下
(instance.props || { tabindex: undefined }).tabindex === undefined

至于写法一和写法二,为什么会出现 instance.props对象 没有和有 tabindex属性
主要是因为创建组件实例对象时createComponentInstance -> normalizePropsOptions(这个方法会根据prop有没有default属性,把prop key加入needCastKeys栈中)
然后setupComponent -> initProps -> setFullProps(这个方法中会根据needCastKeys中的key给props添加属性)
所以就出现了props有和没有tabindex的现象就是因为一个给了default属性一个没给default属性
这是小的区别

说了这么多，就是感觉写法一和写法二都可以
按照您说的，不给的tabindex的属性，就不用给触发器添加这个属性，防止聚焦和键盘导航
当用户提供该属性时，才会给触发器添加该属性

Reply:
太详细了，这么晚回复非常抱歉，最近有些事情比较忙，其他的我没什么问题了。我接下来就把这个合掉